examples/custom-matrix-format/CMakeLists.txt

```diff
@@ -8,4 +8,6 @@ if (GINKGO_BUILD_CUDA AND GINKGO_BUILD_OMP)
     target_link_libraries(custom-matrix-format ginkgo)
     target_include_directories(custom-matrix-format PRIVATE
         ${PROJECT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+    # workaround for clang-cuda/g++ interaction
+    set_target_properties(custom-matrix-format PROPERTIES POSITION_INDEPENDENT_CODE ON)
```

Comment:
Just a comment here, in general my understanding is that this makes the code more robust but could create some performance hit. Nonetheless, it feels like position independent code is in general a good idea. I always wondered whether we should make this a default or not. I don't think it's the best place to discuss it, but if anyone has a strong opinion I would like to hear it.

Reply:
This is only an issue because for some reason, g++ expects relocatable code from the CUDA file, but clang-cuda generates static code. For shared/static libraries, I think CMake already takes care of the PIC property.

Reply:
For that issue that's fine, what I meant was generally whether we should or not always put this flag on anyway.

Reply:
I'm not really sure we need it, the only place where I actually had to manually set this was when generating an object library or an intermediate library that was being used to (statically) build a larger library. Do you have a situation in mind where this would be necessary?